File: clients/csharp/Model/PlayerView.cs
namespace AiCup2019.Model
{
    public struct PlayerView
    {
        public int MyId { get; set; }
        public Model.Game Game { get; set; }
        public PlayerView(int myId, Model.Game game)
        {
            this.MyId = myId;
            this.Game = game;
        }
        public static PlayerView ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new PlayerView();
            result.MyId = reader.ReadInt32();
            result.Game = Model.Game.ReadFrom(reader);
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(MyId);
            Game.WriteTo(writer);
        }
    }
}

File: clients/fsharp/Model/PlayerView.fs
namespace AiCup2019.Model

type PlayerView = {MyId: int; Game: Game} with
    member this.writeTo (writer: System.IO.BinaryWriter) =
        writer.Write this.MyId
        this.Game.writeTo writer

    static member readFrom (reader: System.IO.BinaryReader) =
        {
            MyId = reader.ReadInt32()
            Game = Game.readFrom reader
        }
